File: playwright/pages/login_page.py
import re

from utils.url import url


class LoginPage:
    def __init__(self, page, base_url):
        self.page = page
        self.base_url = base_url

    def goto(self):
        self.page.goto(url(self.base_url, "/login"))

    def login(self, email, password):
        self.page.fill("#login_email", email)
        self.page.fill("#login_password", password)

        with self.page.expect_navigation():
            self.page.click("#login_login")

    def onExpectedPage(self, page, expected):
        pattern = rf"/{re.escape(expected)}(/|$)"
        assert re.search(pattern, page.url)

File: playwright/tests/pages/LoginPage.ts
import { Page, expect } from "@playwright/test";

type ExpectedPage = "courtorder" | "org";

/**
 * <FRONT_URL>/login
 */
export default class LoginPage {
  constructor(private page: Page) {}

  async goto() {
    await this.page.goto("/login");
  }

  async login(user: { email: string; password: string }) {
    await this.page.fill("#login_email", user.email);
    await this.page.fill("#login_password", user.password);

    await Promise.all([
      this.page.waitForLoadState("networkidle"),
      this.page.click("#login_login"),
    ]);
  }

  async expectOnPage(expected: ExpectedPage) {
    await expect(this.page).toHaveURL(new RegExp(`/${expected}(/|$)`));
  }
}
